sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/statefulOperators.scala

```diff
@@ -320,11 +327,48 @@ trait StateStoreWriter
    * the driver after this SparkPlan has been executed and metrics have been updated.
    */
   def getProgress(): StateOperatorProgress = {
+    // Still publish instance metrics that are marked with ignoreIfUnchanged,
+    // as those unchanged metric values may contain important information
+    val instanceMetricsToReport = stateStoreInstanceMetrics
+      .filter {
+        case (name, _) =>
+          val metric = longMetric(name)
+          val metricConfig = instanceMetricConfiguration(name)
+          // Keep instance metrics that are updated or aren't marked to be ignored,
+          // as their initial value could still be important.
+          !metric.isZero || !metricConfig.ignoreIfUnchanged
+      }
+      .groupBy {
+        // Group all instance metrics underneath their common metric prefix
+        // to ignore partition and store names.
+        case (name, _) =>
+          (
+            instanceMetricConfiguration(name).stateStoreProvider,
+            instanceMetricConfiguration(name).metricPrefix
+          )
+      }
+      .flatMap {
+        case (_, metrics) =>
+          // Select at most N metrics based on the metric's defined ordering
+          // to report to the driver. For example, ascending order would be taking the N smallest.
```

Comment:
nit: Let's give this information to the interface. Let's assume developers of new metric would only be able to see the interface to begin with.